### PR TITLE
Remove src init file contents

### PR DESCRIPTION
**Pull request by mattdangerw**
I don't think there is any need for this anymore with our explicit API export surface. Less toil to just remove it.

### CODE REVIEW

File: keras_nlp/__init__.py [hunk 1]
```diff
@@ -19,6 +19,14 @@
 
 import os
 
+# sentencepiece is segfaulting on tf-nightly if tensorflow is imported first.
+# This is a temporary fix to restore our nightly testing to green, while we look
+# for a longer term solution.
```
Comment (SamanehSaadat) [inline keras_nlp/__init__.py:24]: Does this comment still stand?

File: keras_nlp/src/version_utils.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Copyright 2023 The KerasNLP Authors
+# Copyright 2021 The KerasNLP Authors
```
Comment (SamanehSaadat) [inline keras_nlp/src/version_utils.py:1]: Do we want to go back in time? :D

File: keras_nlp/src/__init__.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# Copyright 2021 The KerasNLP Authors
+# Copyright 2023 The KerasNLP Authors
```
Comment (SamanehSaadat) [inline keras_nlp/src/__init__.py:1]: Now that we're changing the year, why not 2024?

**Review by SamanehSaadat — APPROVED** [review]
Thanks, Matt!

Just left some mostly year-related comments!

File: keras_nlp/src/version_utils.py (hunk 2 above)
Comment (mattdangerw) [reply keras_nlp/src/version_utils.py:1]: oops yeah was just trying to make the diff here go away, but let me update everything

File: keras_nlp/__init__.py (hunk 1 above)
Comment (mattdangerw) [reply keras_nlp/__init__.py]: i'm not sure if we still need this weird work around or not. i'll leave it in for now, try removing on a separate pr

**Review by divyashreepathihalli — APPROVED** [review]

**Pull request merged**